SPIKES/Routing/go_router_example/lib/pages/home_page.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+class HomePage extends StatelessWidget {
+  final Widget child;
+  const HomePage({Key? key, required this.child}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text("Home page")),
+      body: child,
+      bottomNavigationBar: BottomNavigationBar(
+        items: const [
+          BottomNavigationBarItem(
+              icon: Icon(Icons.home_rounded), label: "Home"),
+          BottomNavigationBarItem(
+              icon: Icon(Icons.menu_rounded), label: "Bottom menu"),
+          BottomNavigationBarItem(
+              icon: Icon(Icons.menu_book_rounded), label: "Bottom Menu 2"),
+        ],
+        currentIndex: _calculateSelectedIndex(context),
+        onTap: (int idx) => _onItemTapped(idx, context),
+      ),
+    );
+  }
+
+  static int _calculateSelectedIndex(BuildContext context) {
+    final GoRouter route = GoRouter.of(context);
+    final String location = route.location;
+    if (location.startsWith('/first')) {
```

Comment:
Instead of looking at the routes, we could also display the capabilities of go route using the :kind(page1|page2|page3) aproach